mflowgen/common/rtl/gen_rtl.sh

```diff
@@ -1,80 +1,86 @@
 #!/bin/bash
-if [ $soc_only = True ]; then
-  echo "soc_only set to true. Garnet not included"
-  touch outputs/design.v
+# Hierarchical flows can accept RTL as an input from parent graph
+if [ -f ../inputs/design.v ]; then
```

Comment:
Can you explain this more ... why up one directory here and two directories in the line below?

Reply:
It's a slight quirk of how you create symlinks. This is running from inside the RTL node. From that directory, the input file is at ../inputs/design.v. But when we create a symlink, the path is relative to where the symlink will be located. Since the symlink is in the outputs directory, we need the path to the input rtl from there, which is ../../inputs/design.v

Reply:
Yeah ln quirks...

In a shell script, I usually like to cd into the destination before symlinking to make it very obvious what is happening. But then you have to cd back... or do it inside a subshell with parentheses "(cd and ln -s)". Technically there is a ln -sr where the 'r' is relative and does the right thing, but not all ln commands have -r so you cannot depend on it.

Reply:
Yes I personally prefer the parens-subshell approach. So in the line below instead of 
`ln -s ../../inputs/design.v outputs/design.v`
you'd simply do
`(cd outputs; ln -s ../../inputs/design.v)`

Reply:
Makes sense.. just updated it.